src/bam/mod.rs

```diff
@@ -434,7 +581,68 @@ impl IndexedReader {
         }
     }
 
-    pub fn fetch(&mut self, tid: u32, beg: u64, end: u64) -> Result<()> {
+    /// Fetch reads to be used either in .read() or .records
+    ///
+    /// The arguments may be anything that can be converted into a FetchDefinition
+    /// such as
+    ///
+    /// * tid: u32 -> fetch everything on this reference
+    /// * reference_name: &[u8] or *&str -> fetch everything on this reference
+    /// * (tid, start, stop): (u32, i64, i64) -> fetch in this region on this tid
+    /// * (reference_name, start, stop): (&[u8]|&str, i64, i64) -> fetch in this region on this tid
+    /// * FetchDefinition::All or "." -> Fetch overything
+    /// * FetchDefinition::Unmapped "*" -> Fetch unmapped (as signified by the 'unmapped' flag in the BAM - might be
+    ///   unreliable with some aligners.
+    ///
+    /// This replaces the old fetch and fetch_str implementations.
+    pub fn fetch<'a, T: Into<FetchDefinition<'a>>>(&mut self, fetch_definition: T) -> Result<()> {
+        //this 'compile time redirect' safes us
+        //from monomorphing the 'meat' of the fetch function
+        self._inner_fetch(fetch_definition.into())
+    }
```

Comment:
This doc comment should have at least one usage example I think! Or even multiple examples to showcase the different variants

Reply:
Added. The example-boilerplate to actual fetch calls ratio doesn't warrant multiples though.